l3build-arguments.lua

```diff
@@ -34,7 +34,7 @@ local insert           = table.insert
 
 -- Parse command line options
 
-option_list =
+A.option_list =
```

Comment:
Deosn't this make `option_list` purely local? How then do we add to it from a `build.lua` file?

Reply:
The problem is that a global `option_list` may be edited in `build.lua`. Is it what is expected ?
If so `option_list` should not be relied upon before `build.lua` is read.
The idea is to define a bunch of private variables making them local, then expose only some of them globally either RW or RO, and then import `build.lua`. Then we read only what we expect.

Reply:
The original syntax was `texlua build.lua` rather than `l3build`, which meant that `build.lua` was read _before_ `l3build.lua`. That's still supported by deprecated. As such, everything is (supposed to be) set up such that you can read `build.lua` either before or after `l3build.lua`. That's why all of `l3build.variables` is of the form `foo = foo or {<init>}`.